signing_test.go

```diff
@@ -154,11 +156,22 @@ func TestRoundtripsJWSCorruptSignature(t *testing.T) {
 // TestSignerWithBrokenRand tests that using a broken random reader with PSS
 // signature algorithms returns a valid error.
 func TestSignerWithBrokenRand(t *testing.T) {
-	// As of go1.20, the random parameter used in rsa.SignPKCS1v15 is legacy and
-	// ignored, and it can be nil meaning that there's no broken random-ness test
-	// applicable for signing RS256, RS384, or RS512. Instead, focus on the PSS
-	// signature algorithms.
-	pssSigAlgs := []SignatureAlgorithm{PS256, PS384, PS512}
+	// As of go1.20, the random input parameter used in rsa.SignPKCS1v15 is
+	// legacy and ignored, and it can be nil meaning that there's no broken
+	// random-ness test applicable for signing RS256, RS384, or RS512.
+	sigAlgs := []SignatureAlgorithm{PS256, PS384, PS512}
+
+	// We still need to test that users building/testing go-jose on older
+	// versions of go will return an error if the random reader is broken.
+	goVer := runtime.Version()
+	goVer = strings.Trim(goVer, "go")
+	match, err := regexp.MatchString(`1.1\d+(\.\d+)?`, goVer)
+	if err != nil {
+		t.Errorf("error while parsing regex for go version")
+	}
+	if match {
```

Comment:
what if this is just
```suggestion
	if strings.HasPrefix(runtime.Version(), "go1.1") {
```